### PR TITLE
Refactoring from latest reviews + gets testing coverage up to 100%

### PR DESCRIPTION
**Pull request by NelsonGaldeman**
I've went through the open reviews and updated what was missing + some more things I spotted (https://github.com/ImpeccableHQ/swapr-lp-zapping/issues/1 & https://github.com/ImpeccableHQ/swapr-lp-zapping/issues/3)

Any questions please lmk.

### CODE REVIEW

File: contracts/Zap.sol [hunk 1]
```diff
@@ -291,24 +291,24 @@ contract Zap is Ownable, ReentrancyGuard {
...
-                TransferHelper.safeTransfer(tokens[i], msg.sender, tokenBal);
+                TransferHelper.safeTransfer(tokens[i], msg.sender, tokenBalance);
             }
         }
     }
 
     /** 
     @notice Pause the contract
     */
-    function toggleContractActive() public onlyOwner {
```
Comment (berteotti) [inline contracts/Zap.sol:311]: Why changing from `public` to `external`?

File: contracts/Zap.sol [hunk 2]
```diff
@@ -291,24 +291,24 @@ contract Zap is Ownable, ReentrancyGuard {
     @param tokens Tokens' addresses transferred to the msg sender as affiliate share of protocol fee
     */
     function affilliateWithdraw(address[] calldata tokens) external {
-        uint256 tokenBal;
+        uint256 tokenBalance;
```
Comment (berteotti) [inline contracts/Zap.sol:283]: Like the readability

File: contracts/Zap.sol [hunk 3]
```diff
@@ -183,7 +182,19 @@ contract Zap is Ownable, ReentrancyGuard {
...
+            // unwrap to native currency
+            IWETH(nativeCurrencyWrapper).withdraw(amountTo);
+            totalProtocolFeePortion = _subtractProtocolFee(nativeCurrencyAddress, amountTo, affiliate);
+            TransferHelper.safeTransferETH(receiver, amountTo - totalProtocolFeePortion);
+        } else {
+            totalProtocolFeePortion = _subtractProtocolFee(tokenTo, amountTo, affiliate);
+            TransferHelper.safeTransfer(tokenTo, receiver, amountTo - totalProtocolFeePortion);
+        }
+
+        amountTransferred = amountTo - totalProtocolFeePortion;
```
Comment (berteotti) [inline contracts/Zap.sol]: Why is it better to move the code and delete the function?

**Review by berteotti — COMMENTED** [review]
Just curious about this refactoring :)

File: contracts/Zap.sol (hunk 1 above)
Comment (NelsonGaldeman) [reply contracts/Zap.sol:311]: It's not being used internally (so no point on being public over external) and it's cheaper gas-cost wise to execute as external

File: contracts/Zap.sol (hunk 3 above)
Comment (NelsonGaldeman) [reply contracts/Zap.sol]: Gas costs mainly, if it were being used in more places I would be in favour of readability/maintainability/less-code-repetition but it's not the case.

Comment (NelsonGaldeman) [reply contracts/Zap.sol]: Also, it's a quite short function so it doesn't affect readability/undestanding much...

**Comment by fnanni-0** [comment]
Review + questions about the smart contract design: https://docs.google.com/document/d/14Z65AMQcZPqekymPcHuZ76kSxrX7kteNPjDYwaQK7H0/edit?usp=sharing.

**Comment by NelsonGaldeman** [comment]
Thanks for the review @fnanni-0! I've just added changes accodingly. I've ignored fee related comments as the whole model is changing on #6 (we can discuss your points there if you want though)

**Comment by fnanni-0** [comment]
LGTM!

**Review by AugustoL — APPROVED** [review]

**Pull request merged**